ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/NullScanTaskDispatcher.java

```diff
@@ -159,21 +165,46 @@ private void processTableScans(MapWork work, Set<TableScanOperator> tableScans)
           } else {
             managedEmptyPathMap.put(path, false);
           }
+        } else {
+          numberOfUnmanagedPaths++;
         }
       }
     }
 
+    int fetcherPoolParallelism = Math.min(MAXIMUM_ASYNC_STATUS_FETCHER_COUNT, (int)Math.sqrt(numberOfUnmanagedPaths));
+    ForkJoinPool pool = null;
+    if (fetcherPoolParallelism > 1) { // dont create thread pool for a few partitions
```

Comment:
When HIVE_COMPUTE_SPLITS_NUM_THREADS is used, setting it to 1 would automatically have single thread behaviour. If so, it wouldn't be necessary to do this check?

Reply:
No, in that case lookup will happen in another thread than calling thread. So we would pay the price of creating a new thread for no benefit. This check ensures that we do the lookup in calling thread when creating a new thread is not justified performance wise.